From kristen.c.accardi@intel.com Tue Jan 17 16:54:14 2006
From: Kristen Accardi <kristen.c.accardi@intel.com>
Subject: PCI: really fix parent's subordinate busnr
Cc: <len.brown@intel.com>, <pavel@ucw.cz>
Date: Tue, 17 Jan 2006 16:57:01 -0800
Message-Id: <1137545822.19858.48.camel@whizzy>

After you find the maximum value of the subordinate buses below the child
bus, you must fix the parent's subordinate bus number again, otherwise
it may be too small.

Signed-off-by: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/pci/probe.c |    5 +++++
 1 file changed, 5 insertions(+)

--- gregkh-2.6.orig/drivers/pci/probe.c
+++ gregkh-2.6/drivers/pci/probe.c
@@ -537,6 +537,11 @@ int __devinit pci_scan_bridge(struct pci
 			pci_fixup_parent_subordinate_busnr(child, max);
 			/* Now we can scan all subordinate buses... */
 			max = pci_scan_child_bus(child);
+			/*
+			 * now fix it up again since we have found
+			 * the real value of max.
+			 */
+			pci_fixup_parent_subordinate_busnr(child, max);
 		} else {
 			/*
 			 * For CardBus bridges, we leave 4 bus numbers
